src/containers/ProtocolScreen.js

```diff
@@ -17,9 +17,26 @@ import { getCSSVariableAsNumber } from '../ui/utils/CSSVariables';
   * Check protocol is loaded, and render the stage
   */
 class Protocol extends Component {
+  constructor(props) {
+    super(props);
+    this.interfaceRef = React.createRef();
+  }
+
+  componentDidUpdate(previousProps) {
+    if (previousProps.stage.id !== this.props.stage.id) {
+      this.interfaceRef = React.createRef();
+      this.forceUpdate();
+    }
+  }
+
   // change the stage to the next
   onClickNext = () => {
-    if (!this.props.stage.prompts || this.props.isLastPrompt()) {
+    if (this.getInterfaceRef() && !this.getInterfaceRef().isStageEnding()) {
```

Comment:
I'm not sure what might cause the throw in getInterfaceRef, but should the result be cached in a variable here?

Reply:
Done.